interpreter/CMakeLists.txt

```diff
@@ -163,7 +163,8 @@ if(builtin_llvm)
 
   if(NOT DEFINED LLVM_ENABLE_ASSERTIONS)
     if(CMAKE_BUILD_TYPE MATCHES "(Debug|RelWithDebInfo)"
-     OR LLVM_BUILD_TYPE MATCHES "(Debug|RelWithDebInfo)")
+     OR LLVM_BUILD_TYPE MATCHES "(Debug|RelWithDebInfo)"
+     OR CMAKE_CXX_FLAGS MATCHES "-UNDEBUG")
```

Comment:
Wouldn't the reverse (i.e. not matching `-DNDEBUG`) make more sense here? Someone may have neither `-DNDEBUG` nor `-UNDEBUG` in `CMAKE_CXX_FLAGS`, in which case assertions are enabled, but this won't have any effect.

Reply:
`-O` does not provide `-DNDEBUG` in `CMAKE_CXX_FLAGS`. I.e. IIUC your argument holds but in the opposite way? Anyway:
- if the build is a Debug or RelWithDebInfo build, we get LLVM assertions.
- if not, we get LLVM assertions if `NDEBUG` is explicitly undefined - that's the technique generally used to enable assertions for optimized builds.
What am I missing?

Reply:
If `-UNDEBUG` is not defined, it doesn't mean that `-DNDEBUG` is defined (think user setting flags by hand to `-O2 -march=native`, for example). In that case, assertions should be enabled, but are not because flags won't match `-UNDEBUG` and `CMAKE_BUILD_TYPE` will be unset (due to flags set by hand).

Reply:
I am still lost. Please state a case where the code above does the wrong thing. Taking what you mention here:
Specifying `-O2` will implicitly set `-DNDEBUG`, so no assertions. That's what you'd expect, right? I *guess* passing `-O2` you'd take `Release` as `LLVM_BUILD_TYPE`. So assertions should be disabled, and this PR does not change that.

Reply:
I already did, just configure with `cmake ... -DCMAKE_CXX_FLAGS="-O2 -march=native" ...`, and see that the code above will not enable assertions. Note that I'm not setting a build type, as that would overrule the custom `CXXFLAGS`.

Reply:
To be clear:
```
epsftws ~ $ gcc -O2 -dM -E - </dev/null | grep NDEBUG
epsftws ~ $ gcc -O2 -dM -E -DNDEBUG - </dev/null | grep NDEBUG
#define NDEBUG 1
```

Reply:
O, you're saying this:

> Specifying -O2 will implicitly set -DNDEBUG

is wrong! Fair enough, thanks!